python/ray/serve/_private/client.py

```diff
@@ -107,54 +106,12 @@ def shutdown(self) -> None:
             del self.handle_cache[k]
 
         if ray.is_initialized() and not self._shutdown:
-            ray.get(self._controller.shutdown.remote())
-            self._wait_for_deployments_shutdown()
-
-            ray.kill(self._controller, no_restart=True)
-
-            # Wait for the named actor entry gets removed as well.
-            started = time.time()
-            while True:
-                try:
-                    ray.get_actor(self._controller_name, namespace=SERVE_NAMESPACE)
-                    if time.time() - started > 5:
-                        logger.warning(
-                            "Waited 5s for Serve to shutdown gracefully but "
-                            "the controller is still not cleaned up. "
-                            "You can ignore this warning if you are shutting "
-                            "down the Ray cluster."
-                        )
-                        break
-                except ValueError:  # actor name is removed
-                    break
-
+            try:
+                ray.get(self._controller.graceful_shutdown.remote())
```

Comment:
Let's put a timeout on this so user code doesn't hang if there's an issue with shutdown:
```suggestion
                ray.get(self._controller.graceful_shutdown.remote(), timeout_s=30)
            except TimeoutError:
                logger.warning("Controller failed to shut down within 30s. Check controller logs for more details.")
```